pkg/user/builder.go

```diff
@@ -48,6 +55,15 @@ func (b *Builder) Email(email string) *Builder {
 	return b
 }
 
+func (b *Builder) Password(strPassword string, binPassword []byte) *Builder {
+	if binPassword == nil {
+		b.p = &strPassword
+	} else {
+		b.u.password = binPassword
+	}
+	return b
+}
+
```

Comment:
String password is not used. And slice should be copied before setting them to flelds

Note: `append(a[:0:0], a...)` is copying a slice

```suggestion
func (b *Builder) Password(password []byte) *Builder {
	if len(password) == 0 {
		b.u.password = nil
	} else {
		b.u.password = append(p[:0:0], p...)
	}
	return b
}

```

Reply:
@rot1024 string password is used for creating a new password that is not created yet .... in other words we need to pass an uncoded password and code it during the building process

bin pass is for the infrastructure layer where I need to fetch the stored binary password and pass it to the user model